renderer/brave_wallet/brave_wallet_render_frame_observer.cc

```diff
@@ -45,6 +45,10 @@ void BraveWalletRenderFrameObserver::DidCreateScriptContext(
     js_ethereum_provider_.reset();
     return;
   }
+  // Wallet provider objects won't be generated for third party iframe
+  if (!render_frame()->IsMainFrame() && !render_frame()->GetMainRenderFrame()) {
```

Comment:
Probably better to use `render_frame()->GetWebFrame()->IsCrossOriginToMainFrame()`.

Reply:
That sounds better, fixed and force-pushed